pkg/apis/nmstate/v1alpha1/nodenetworkconfigurationenactment_types.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeNetworkConfigurationEnactmentList contains a list of NodeNetworkConfigurationEnactment
+type NodeNetworkConfigurationEnactmentList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NodeNetworkConfigurationEnactment `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeNetworkConfigurationEnactment is the Schema for the nodenetworkconfigurationenactments API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=nodenetworkconfigurationenactments,shortName=nnce,scope=Cluster
+type NodeNetworkConfigurationEnactment struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Status NodeNetworkConfigurationEnactmentStatus `json:"status,omitempty"`
+}
+
+// NodeNetworkConfigurationEnactmentStatus defines the observed state of NodeNetworkConfigurationEnactment
+// +k8s:openapi-gen=true
+type NodeNetworkConfigurationEnactmentStatus struct {
+	Conditions ConditionList `json:"conditions"`
+}
+
+const (
+	NodeNetworkConfigurationEnactmentConditionAvailable   ConditionType = "Available"
+	NodeNetworkConfigurationEnactmentConditionFailing     ConditionType = "Failing"
+	NodeNetworkConfigurationEnactmentConditionProgressing ConditionType = "Progressing"
+)
+
+const (
+	NodeNetworkConfigurationEnactmentConditionFailedToConfigure        ConditionReason = "FailedToConfigure"
+	NodeNetworkConfigurationEnactmentConditionSuccessfullyConfigured   ConditionReason = "SuccessfullyConfigured"
+	NodeNetworkConfigurationEnactmentConditionConfigurationProgressing ConditionReason = "ConfigurationProgressing"
+)
+
+func EnactmentKey(node string, policy string) types.NamespacedName {
+	return types.NamespacedName{Name: fmt.Sprintf("%s.%s", node, policy)}
+}
+
+func NewEnactment(node corev1.Node, policy NodeNetworkConfigurationPolicy) NodeNetworkConfigurationEnactment {
+	return NodeNetworkConfigurationEnactment{
```

Comment:
Would it make sense to initialize with all our conditions set to Unknown? In the API, a missing condition is considered to be Unknown implicitly, but I feel it may be better for the user to always give them all conditions.

Reply:
Yep, I prefer explicit.